src/order/sup_indep.lean

```diff
@@ -65,6 +65,40 @@ lemma sup_indep_iff_disjoint_erase [decidable_eq ι] :
 ⟨λ hs i hi, hs (erase_subset _ _) hi (not_mem_erase _ _), λ hs t ht i hi hit,
   (hs i hi).mono_right (sup_mono $ λ j hj, mem_erase.2 ⟨ne_of_mem_of_not_mem hj hit, ht hj⟩)⟩
 
+@[simp] lemma sup_indep_pair [decidable_eq ι] {i j : ι} (hij : i ≠ j) :
+  ({i, j} : finset ι).sup_indep f ↔ disjoint (f i) (f j) :=
```

Comment:
Wouldn't this be faster with a `sup_indep_insert` lemma?

Reply:
What would that lemma state?

Reply:
That `insert a s` (or `s.cons a h` to avoid `decidable_eq`) is `sup_indep` iff `disjoint (f a) (s.sup f)` and for all `t ⊆ s` and `i ∉ t`, `disjoint (f i) (t.sup f ⊔ f a)`.

Reply:
I thought about the `cons` version for `pair`, but found it annoying to state because I had to construct the proof as part of the statement.

That `insert` lemma doesn't seem super useful, as it doesn't actually help once you get to `{a, b, c}`, as it can't be applied twice.